src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -201,15 +201,23 @@ private static bool TryGetDefaultAddress(int pid, out string defaultAddress)
 
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                defaultAddress = $"dotnet-diagnostic-{pid}";
+                if (File.Exists($"{IpcRootPath}dotnet-dsrouter-diagnostic-{pid}"))
```

Comment:
These feel like an over-optimization for dsrouter scenarios that impacts all diagnostics scenarios. In my opinion, all diagnostic tools should not have to scan two addresses spaces when discovering processes to diagnose.

This also has the issue of if dsrouter leaves behind its custom diagnostic port Unix domain socket file, then that process ID is forever inaccessible until the file is cleared out. For example, dsrouter runs as process ID 12345, then is killed (crashes, etc) leaving behind the file, then another non-dsrouter process starts as process ID 12345; this process cannot be discovered due to the proposed algorithm.

It might be better to leave this alone and then add an ability to the diagnostic tools to reach out a specific address for connection. For example, the tools can be provided the port name and the mode in which they connect e.g. `dotnet-counters --dignostic-port dotnet-dsrouter-diagnostic-{pid} --diagnostic-port-mode connect`. I know that this doesn't work well for Linux given the port has an "arbitrary" disambiguator, but I'm proposing a general concept rather than a specific implementation.

Reply:
yes, part of this PR is to have the discussion, @josalem and I discussed the ability to open up to explicit use named ipc ports in all tooling, dsrouter can already do that using -ipcs argument, so if we are ok adding the argument I'm perfectly happy to do that way instead. Just one side effect, then dsrouter will have issues using the --process-id/--name argument since it won't bind the default port, but we could always use the option to disable diagnostics in dsrouter process and detect that case and warn as a fallback.

Reply:
If we are OK doing that, I can adjust this PR tomorrow.

Reply:
My 2 cents ...
I think we are better off making the dsrouter proxy connection something that a tool user configures explicitly rather than trying to trying to broaden the meaning of the --process-id/--name arguments. For example the workflow might look like this:
```
# Run the proxy
> dsrouter some_args
Proxy is listening for connection at address: /tmp/android_proxy.sock
To connect local dotnet tools use argument: --diagnostic-port /tmp/android_proxy.sock,connect

# Run a tool
> dotnet-trace --diagnostic-port /tmp/android_proxy.sock,connect ...
```

Reply:
@noahfalk Thanks for feedback, please check my long comment below, but I'm landing on something similar, either having a explicit --connect option that can go with --diagnostic-port, making it explicit to connect to a custom named IPC channel. Not sure if we preferer the runtime way backing it all into the --diagnostic-port argument or having it as an explicit option. So either we do it like you propose, `--diagnostic-port /tmp/android_proxy.sock,connect` or we do it, `--diagnostic-port /tmp/android_proxy.sock --connect`.

Reply:
Another variation to consider `--connect /tmp/android_proxy.sock`. Personally I like it because it is shorter but I know these can be very subjective so I'd love to get some other opinions from you guys.

Reply:
So then we reserve the --diagnostic-port for reverse connect, and the new `--connect` to connect against a custom named IPC channel, I'm fine with that.